eos_data_distribution/ndn/http.py

```diff
@@ -85,24 +84,34 @@ def get_last_modified(headers):
 
 class Getter(object):
 
-    def __init__(self, url, onData, session=None, chunk_size=chunks.CHUNK_SIZE):
+    def __init__(self, url, onData, session=None, chunk_size=chunks.CHUNK_SIZE, onHeaders=None):
         super(Getter, self).__init__()
 
         self.url = url
         self.onData = onData
         self.chunk_size = chunk_size
+        self.onHeaders = onHeaders
 
         self._session = session
         if self._session is None:
             self._session = make_soup_session()
 
         # XXX -- this is a bit ugly that we're making an HTTP request
         # in the constructor here...
-        self._headers = fetch_http_headers(self._session, self.url)
+        self._headers = fetch_http_headers(
+            self._session, self.url, self.got_http_headers)
+        logger.debug('getter init: %s', url)
+
+    def got_http_headers(self, o, msg, data=None):
+        logger.debug('got HTTP HEADERS: %s', msg.response_headers)
+        self._headers = msg.response_headers
+
         self._size = get_content_size(self._headers)
         if self._size == -1:
             raise ValueError("Could not determine Content-Size for %s" % url)
```

Comment:
There’s no way for this `ValueError` to be caught. It should really be passed to `onHeaders` as a failure value, I think? Not entirely sure of the Pythonic way of doing this.